src/llama_stack_client/lib/direct/direct.py

```diff
@@ -18,15 +20,30 @@
 
 class LlamaStackDirectClient(LlamaStackClient):
     def __init__(self, config: StackRunConfig, **kwargs):
+        raise TypeError("Use from_yaml() or from_template() instead of direct initialization")
+
+    @classmethod
+    async def from_config(cls, config: StackRunConfig, **kwargs):
```

Comment:
one more comment. in this code path, where is env var replacement happening? is it happening up in the stack?